tsrc/test/cli/test_init.py

```diff
@@ -265,3 +265,26 @@ def test_several_remotes(
     )
     assert rc == 0, output
     assert output == "git@upstream.com"
+
+
+def test_singular_remote(
+    tsrc_cli: CLI, git_server: GitServer, workspace_path: Path
+) -> None:
+    foo_url = git_server.add_repo("foo")
+    # fmt: off
+    git_server.manifest.set_repo_remotes(
+        "foo",
+        [("origin", foo_url),
+         ("upstream", "git@upstream.com")])
+    # fmt: on
+
+    # only use "origin" remote
+    tsrc_cli.run("init", git_server.manifest_url, "-r", "origin")
+
+    foo_path = workspace_path / "foo"
+    rc, output = tsrc.git.run_captured(
+        foo_path, "remote", "show", check=False
```

Comment:
Instead of using `check=False` and asserting that return code is 0, you can also use `check=True`

That way the test will fail with a nicer error